shutdown_test.go

```diff
@@ -87,6 +87,59 @@ func TestShutdown(t *testing.T) {
 		assert.NotNil(t, <-done1, "done channel 1 did not receive signal")
 		assert.NotNil(t, <-done2, "done channel 2 did not receive signal")
 	})
+
+	t.Run("shutdown app with non-zero, user-specified exit code with app.Run()", func(t *testing.T) {
+		t.Parallel()
+
+		var (
+			exited   bool
+			exitCode int
+		)
+		exit := func(code int) {
+			exited = true
+			exitCode = code
+		}
+
+		var s fx.Shutdowner
+		expectedExitCode := 2
+		shutdown := func(s fx.Shutdowner) error {
+			return s.Shutdown(fx.WithExitCode(expectedExitCode))
+		}
+
+		app := fxtest.New(
+			t,
+			fx.Populate(&s),
+			fx.WithExit(exit),
+			fx.Invoke(shutdown),
```

Comment:
This is incorrect usage per our documentation of Shutdowner:

```
// In practice this means Shutdowner.Shutdown should not be called from an
// fx.Invoke, but from a fx.Lifecycle.OnStart hook.
```

Previously, this just wouldn't work.
That has likely changed since #805 so we can update the documentation,
but all usages of Shutdowner up until this point are inside a lifecycle hook (on or after App.Start),
so we need to test for that too.